consensus_decentralization/aggregate.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import logging
+from collections import defaultdict
+import consensus_decentralization.helper as hlp
+from consensus_decentralization.map import ledger_mapping
+
+
+class Aggregator:
+    """
+    Class used to aggregate the results of the mapping process.  This class then reads a json file like
+    that and aggregates the results for a given timeframe (e.g. month) by counting the number of blocks produced by
+    each entity. The result is a dictionary of entities that produced blocks in the given timeframe and the number of
+    blocks they produced
+    """
+
+    def __init__(self, project, io_dir, data_to_aggregate):
+        """
+        :param project: str. Name of the project
+        :param io_dir: Path. Path to the project's output directory
+        """
+        self.project = project
+        self.data_to_aggregate = data_to_aggregate
+        self.aggregated_data_dir = io_dir / 'blocks_per_entity'
+        self.aggregated_data_dir.mkdir(parents=True, exist_ok=True)
+
+    def aggregate(self, timeframe):

```

Comment:
I find it a bit confusing to name both functions in the same file `aggregate`. You could rename this one `process`. Don't have strong feelings though, so leaving it like that is also fine.